src/Agg.cpp

```diff
@@ -53,6 +54,15 @@ namespace geopm
         return result;
     }
 
+    double Agg::difference(const std::vector<double> &operand)
```

Comment:
If you want to avoid repeating the stubtracting lambda, maybe you can add another CombinedSignal derived type (e.g. 'DifferenceCombinedSignal') and override the sample() method.  This would mean that the CombinedSignal base class with injected function is never used, which might be a good thing.

Reply:
I originally went down the path of creating a DifferenceCombined signal subclass, and didn't like the result.  It was a bit more code to do it that way.  I think there are other spots where we're using the CombinedSignal base class with the constructor default of Agg::sum and with the injected function (e.g. in push_combined_signal) so I'm not sure we can get away from calling CombinedSignal this way entirely.  Adding the Agg function resulted in more clean/tidy code in my opinion.

Agreed that for the combined signal of ```TEMPERATURE``` the ordering of the signals is important for this to work.  This new Agg::difference function will work OK for any size vector.  With the present example of ```TEMPERATURE``` it only makes sense with 2 elements, but I could see a future use case where we try to calculate left over power with something like ```TDP_POWER - PACKAGE_POWER - DRAM_POWER``` or similar.  I think there's value in adding this Agg function for those situations (though they may be rare).

I haven't come across any cases in my testing where the lambda fixed a problem with Agg::difference.